tests/jballerina-unit-test/src/test/resources/test-src/types/table/record-type-table-key.bal

```diff
@@ -201,6 +201,53 @@ function testVariableNameFieldAsKeyField() {
     assertEquality(expected, tb.toString());
 }
 
+function testDefaultValueFieldAsKeyField() {
+    string expected = "[{\"id\":1,\"name\":\"Jo\"},{\"id\":2,\"name\":\"Amy\"}]";
+    table<record {readonly int id = 1; string name;}> key (id) tb = table [
+        {id, name: "Jo"},
+        {id: 2, name: "Amy"}
+    ];
+    assertEquality(expected, tb.toString());
+}
+
+type CustomRecord record {
+    readonly int id = 1;
+    string name;
+};
+
+function testDefaultValueFieldAsKeyField2() {
+    string expected = "[{\"id\":1,\"name\":\"Jo\"},{\"id\":2,\"name\":\"Amy\"}]";
+    table<CustomRecord> key (id) tb = table [
+        {id, name: "Jo"},
+        {id: 2, name: "Amy"}
+    ];
+    assertEquality(expected, tb.toString());
+}
+
+type CustomRecord2 record {
+    readonly int id = 1;
+    readonly string status = "status - 1";
+    string name;
+};
+
+const string status = "status - 1";
+
+function testDefaultValueFieldAsKeyField3() {
+    string expected =
+        "[{\"id\":1,\"status\":\"status - 1\",\"name\":\"Jo\"},{\"id\":2,\"status\":\"status - 2\",\"name\":\"Amy\"}]";
+    table<CustomRecord2> key (id, status) tb = table [
+        {id, status, name: "Jo"},
+        {id: 2, status: "status - 2", name: "Amy"}
+    ];
+    assertEquality(expected, tb.toString());
+}
```

Comment:
```
type CustomRecord2 record {
    readonly int id;
    readonly string status;
    string name;
};

const string status = "status - 1";
const int id = 1;

public function main() {
    table<CustomRecord2> key (id, status) tb = table [
        {id, status, name: "Jo"},
        {id: 2, status: "status - 2", name: "Amy"}
    ];
}
```
This is works as same mentioned here, So, is there any use of initializing with `default` values?  
Can't we use the default values to initialize the uninitialized fields `id`, `status`?

Reply:
Our issue is manly focused on using default value record field as a `key` in table descriptor. So in this PR mostly focus on key attributes.

Reply:
Yes. I will add tests.